rskj-core/src/test/java/co/rsk/peg/federation/FederationChangeIT.java

```diff
@@ -0,0 +1,420 @@
+            .withLockingCapSupport(lockingCapSupport)
+            .build();
+    }
+
+    private static void addNewBtcBlockOnTipOfChain(BtcBlockStore blockStore) throws Exception {
+        var chainHead = blockStore.getChainHead();
+        var btcBlock = new BtcBlock(
+            BRIDGE_CONSTANTS.getBtcParams(),
+            1,
+            chainHead.getHeader().getHash(),
+            BitcoinTestUtils.createHash(chainHead.getHeight() + 1),
+            0,
+            0,
+            0,
+            List.of());
+        var storedBlock = new StoredBlock(
+            btcBlock,
+            BigInteger.ZERO,
+            chainHead.getHeight() + 1
+        );
+
+        blockStore.put(storedBlock);
+        blockStore.setChainHead(storedBlock);
+    }
+
+    private List<UTXO> createRandomUTXOs(Address owner) {
+        Script outputScript = ScriptBuilder.createOutputScript(owner);
+        List<UTXO> result = new ArrayList<>();
+
+        int howMany = getRandomInt(50, 500);
+        for (int i = 1; i <= howMany; i++) {
+            Coin randomValue = Coin.valueOf(getRandomInt(10_000, 1_000_000_000));
+            Sha256Hash utxoHash = BitcoinTestUtils.createHash(i);
+            result.add(new UTXO(utxoHash, 0, randomValue, 0, false, outputScript));
+        }
+
+        return result;
+    }
+    
+    private int getRandomInt(int min, int max) {
+        return TestUtils.generateInt(FederationChangeIT.class.toString() + min, max - min + 1) + min;
+    }
+    
+    /* Assert federation change related methods */
+
+    private void assertUTXOsMovedFromNewToOldFederation(List<UTXO> originalUTXOs, ActivationConfig.ForBlock activations) {
+        // Assert old federation exists in storage and matches
+        assertNotNull(
+            federationStorageProvider.getOldFederation(BRIDGE_CONSTANTS.getFederationConstants(), activations));
+        // Assert new federation exists in storage and matches
+        assertNotNull(
+            federationStorageProvider.getNewFederation(BRIDGE_CONSTANTS.getFederationConstants(), activations));
+        // Assert old federation holds the original utxos
+        List<UTXO> utxosToMigrate = federationStorageProvider.getOldFederationBtcUTXOs();
+        assertTrue(originalUTXOs.stream().allMatch(utxosToMigrate::contains));
+        // Assert the new federation does not have any utxos yet
+        assertTrue(federationStorageProvider
+            .getNewFederationBtcUTXOs(BRIDGE_CONSTANTS.getBtcParams(), activations)
+            .isEmpty());
+    }
+
+    private void assertNewFederationIsActive(Address newFederationAddress, Address oldFederationAddress) {
+        // New active and retiring federation
+        assertEquals(newFederationAddress, bridgeSupport.getActiveFederationAddress());
+        assertEquals(oldFederationAddress, bridgeSupport.getRetiringFederationAddress());
+    }
+    
+    private void assertNoMigrationHasStarted() throws Exception {
```

Comment:
```suggestion
    private void assertFundsWereNotMigrated() throws Exception {
```

nit, but i think is clearer

Reply:
Updated